packages/block-editor/src/components/block-patterns-paging/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	Button,
+} from '@wordpress/components';
+import { __, _x, _n, sprintf } from '@wordpress/i18n';
+
+export default function Pagination( {
+	currentPage,
+	numPages,
+	changePage,
+	totalItems,
+} ) {
+	return (
+		<VStack>
+			<Text variant="muted">
+				{
+					// translators: %s: Total number of patterns.
+					sprintf(
+						// translators: %s: Total number of patterns.
+						_n( '%s item', '%s items', totalItems ),
+						totalItems
+					)
+				}
+			</Text>
+			<HStack
+				expanded={ false }
+				spacing={ 3 }
+				justify="flex-start"
+				className="block-editor-patterns__grid-pagination"
+			>
+				<HStack expanded={ false } spacing={ 1 }>
+					<Button
+						variant="tertiary"
+						onClick={ () => changePage( 1 ) }
+						disabled={ currentPage === 1 }
```

Comment:
I wonder where we should move the focus after navigating to a different page. Should we retain the focus on the prev/next button? Or should we move the focus to the first pattern to match the visual jump? If we decide to retain the focus then we probably want to change this to `aria-disabled` to prevent focus lost when pressing this button for instance. This can all be reviewed by an accessibility expert though, I'm just tossing out some questions 😅 .

Reply:
I think it would be good to look at this as a follow-up combined with the site editor pagination - ideally we should finalise the `pattern` we should use for this and see if we can have a generic pagination component that can be shared between site/post editor.